src/modules/dataman/dataman.c

```diff
@@ -949,7 +949,15 @@ _file_initialize(unsigned max_offset)
 		close(dm_operations_data.file.fd);
 
 		if (incompat) {
+#ifdef CONFIG_ARCH_BOARD_AEROCORE2
```

Comment:
@smithandrewc 

Ahh - yikes - this must have been broken in the recent work done on dataman!

@LorenzMeier - where is the preferred storage of waypoints now on the ramtron or the sdcard?

I think the issue is that the  ```/fs/mtd_waypoints``` is not a file it is a character device created on the block device.  In this case unlink is not the correct thing to do to a character device (driver).  The erase is done by writing 0xff to the device as in ```mtd_erase```

I imagine when the file name contains "mtd_" (or use stat() and look for buf->st_mode & S_IFCHR).)
we could call ```mtd_erase(k_data_manager_device_path, 1)``` instead of unlink.